backend/src/permissions/permissions.service.spec.ts

```diff
@@ -171,6 +181,27 @@ describe('PermissionsService', () => {
     return note;
   }
 
+  function spyOnNote(methodName: MethodType<typeof noteRepo>) {
+    return jest.spyOn(noteRepo, methodName);
+  }
+  function spyOnEventEmitter(methodName: MethodType<typeof eventEmitter>) {
+    return jest.spyOn(eventEmitter, methodName);
+  }
+
+  function mockNoteMethods(
+    methodName: MethodType<typeof noteRepo>,
+    implementation: ImplementationType,
+  ) {
+    return spyOnNote(methodName).mockImplementationOnce(implementation);
+  }
+
+  function mockEventEmitterMethods(
+    methodName: MethodType<typeof eventEmitter>,
+    implementation: ImplementationType,
+  ) {
+    return spyOnEventEmitter(methodName).mockImplementationOnce(implementation);
+  }
+
```

Comment:
This extraction is a bit overengineered. You use each of this functions for exactly one purpose which would be okay if the extracted code is very complex or long. But in this case it's not really helpful :/

Reply:
@mrdrogdrog I have removed unnecessary abstraction.